src/commands/topics.ts

```diff
@@ -166,6 +168,71 @@ async function editTopicConfig(topic: KafkaTopic): Promise<void> {
   editConfigForm.onDidDispose(() => disposable.dispose());
 }
 
+async function produceMessageFromFile(topic: KafkaTopic) {
+  if (!topic) {
+    vscode.window.showErrorMessage("No topic selected.");
+    return;
+  }
+  const options: vscode.OpenDialogOptions = {
+    canSelectMany: false,
+    openLabel: "Select JSON file",
+    filters: {
+      "JSON files": ["json"],
+    },
+  };
+
+  const fileUri = await vscode.window.showOpenDialog(options);
+  if (fileUri && fileUri[0]) {
+    const filePath = fileUri[0].fsPath;
+    const message = JSON.parse(fs.readFileSync(filePath, "utf-8"));
+
+    if (!message.key || !message.value || message.headers.length === 0) {
```

Comment:
We may want to update this header check for just `!message.headers` since it may not have a `length` property from a JSON.parse (in the event someone's document has a non-iterable `headers` property)